app/models/environment.rb

```diff
@@ -25,6 +26,15 @@ def active_lock
     locks.active.first
   end
 
+  # Configures this environment to auto deploy the given branch.
+  def configure_auto_deploy(branch, options = {})
+    self.update_attributes!(auto_deploy_branch: branch, auto_deploy_user: options[:user])
```

Comment:
The approach I've taken previously sorta cascades in a way which tends to give you greater specificity as teams care more and more about attribution.

In a normal team situation @turadg, @ejholmes, and I are hacking on a fun project. @turadg turns on auto-deployment in slashdeploy. @turadg makes a pr, @atmos piles on some commits, everything looks cool. @ejholmes merges the PR.

* Bot/Deployer : In these cases you have a bot user that creates deployments using a personal OAuth token. This handles the simpler case where someone configures everything for the group. In the above example none of the users are attributed w/ the deployment, the bot is.
* "Have a configured account handle API interaction"(this approach) : This sets up the creation of deployments to be from a specific user that automated tools always attribute to whoever configured it. In the above example @turadg is attributed w/ the deployment. I introduced a bunch of changes after them and @ejholmes approved it. If you ever have to go back and audit it, it's tough to track down who is responsible.
* "Properly attribute git authors" : This is what most businesses want. Since we can(mostly) look up the user by login we can find them and use their token to trigger the deployment and get proper attribution. If we can't find the user, it just falls through to the previous options.

Other stuff people will want is 2fa. We're using yubikeys at heroku and passing them through at the end of the deploy commands. That gets encrypted/decrypted through the payloads and funneled through to the appropriate providers. You get a small window to do actions without it but it's interesting to think about the three examples above when you want to introduce 2fa and what the security model looks like to acceptably bypass it.

> Would be nice to have the option to just pass it a github access token and use that instead.

I think it'd great to have this be a separate path in the app or whatever you could go through the OAuth flow rather than people copying and pasting stuff. "Configure Bot Account" button, or something.